File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def solution(n, k):
    answer = 0
    
    k_num = ""
    
    ## k진수 변환
    while n>0:
        k_num += str(n%k)
        n//=k
        
    k_num = k_num[::-1]
    
    def is_prime(n):
        if n<2: return False
        
        for i in range(2, (int)(n**(1/2))+1):
            if n%i == 0: return False
        
        return True
    
    prime = ""
    for s in k_num:
        if s == '0':
            if prime != "" and is_prime(int(prime)):
                answer += 1
            prime = ""
        else:
            prime += s
            
    if prime != "" and is_prime(int(prime)): answer += 1
    
    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        
        StringBuilder sb = new StringBuilder();
        
        // k진수 변환
        while(n > 0){
            sb.append(String.valueOf(n%k));
            n /= k;
        }
        
        StringBuilder prime = new StringBuilder();
        
        for(char c: sb.reverse().toString().toCharArray()) {
            if(c == '0'){
                if(prime.toString().length() != 0 && isPrime(Long.parseLong(prime.toString()))) answer++;
                prime = new StringBuilder();
            } else {
                prime.append(c);
            }
        }

        if(prime.toString().length() != 0 && isPrime(Long.parseLong(prime.toString()))) answer++;
        
        return answer;
    }
    
    public boolean isPrime(long n){
        if (n<2) return false;
        
        long sqrt = (int)Math.sqrt(n);
        for(long i=2; i<=sqrt; ++i){
            if(n%i == 0){
                return false;
            }
        }
        return true;
    }
}